dashboard/src/modules/components/TableOfContent/index.jsx

```diff
@@ -1,384 +1,104 @@
 import "./index.less";
 
 import {
-  AngleLeftIcon,
-  DownloadIcon,
-  FolderIcon,
-} from "@patternfly/react-icons";
-import {
-  BadgeToggle,
-  Breadcrumb,
-  BreadcrumbHeading,
-  BreadcrumbItem,
   Divider,
-  DrilldownMenu,
-  Dropdown,
-  DropdownItem,
-  Menu,
-  MenuBreadcrumb,
-  MenuContent,
-  MenuItem,
-  MenuList,
+  Flex,
+  FlexItem,
+  List,
+  ListItem,
+  Sidebar,
+  SidebarContent,
+  SidebarPanel,
 } from "@patternfly/react-core";
-import React, { useEffect, useState } from "react";
-import {
-  TableComposable,
-  Tbody,
-  Td,
-  Th,
-  Thead,
-  Tr,
-} from "@patternfly/react-table";
-import {
-  fetchTOC,
-  updateContentData,
-  updateCurrData,
-  updateSearchSpace,
-  updateStack,
-  updateTableData,
-} from "actions/tableOfContentActions";
+import React, { useEffect } from "react";
+import { fetchTOC, setActiveFileContent } from "actions/tocActions";
 import { useDispatch, useSelector } from "react-redux";
 
-import { DEFAULT_PER_PAGE } from "assets/constants/paginationConstants";
-import { EmptyTable } from "../TableComponent/common-components";
-import { SearchTOC } from "./common-components";
-import TablePagination from "../PaginationComponent";
+import { DownloadIcon } from "@patternfly/react-icons";
+import DrilldownMenu from "./DrillDownMenu";
 import { useParams } from "react-router";
 
 const TableOfContent = () => {
-  const { endpoints } = useSelector((state) => state.apiEndpoint);
-  const [menuDrilledIn, setMenuDrilledIn] = useState([]);
-  const [drilldownPath, setDrillDownPath] = useState([]);
-  const [activeMenu, setActiveMenu] = useState("rootMenu");
-  const [breadCrumb, setBreadCrumb] = useState(null);
-  const [activeFile, setActiveFile] = useState(null);
-  const [breadCrumbLabels, setBreadCrumbLabels] = useState([]);
-  const [param, setParam] = useState("");
-  const [page, setPage] = useState(1);
-  const [perPage, setPerPage] = useState(DEFAULT_PER_PAGE);
-  const params = useParams();
   const dispatch = useDispatch();
-  let dirCount = 0;
-  let fileCount = 0;
+  const params = useParams();
+
+  const { endpoints } = useSelector((state) => state.apiEndpoint);
+  const { activeFile, inventoryDownloadLink } = useSelector(
+    (state) => state.toc
+  );
   useEffect(() => {
     if (Object.keys(endpoints).length > 0)
-      dispatch(fetchTOC(params["dataset_id"], "", false));
+      dispatch(fetchTOC(params["dataset_id"], ""));
   }, [dispatch, endpoints, params]);
```

Comment:
I believe that the fourth parameter (`isSubdir`) should be `false`, as it was in the previous code.  I believe that the value of the third parameter (`item`) won't be used (because, with `isSubdir` false, references to it will be skipped over), so you just need a placeholder for it.  So, something like
```javascript
      dispatch(fetchTOC(params["dataset_id"], "", undefined, false));
```
should suffice.

(I believe the reason why this works today is that Javascript provides `undefined` as the value for each omitted parameter, and, in a boolean context, `undefined` works like `false`, and so we get where we need to be; however, that seems like coincidence, more or less, and so we really should be providing all the values which matter (i.e., `isSubdir`).  If you want, you could reorder the parameters so that `item` is the last one, and then you could omit its value here after specifying `isSubdir`, but I think it's better to just provide all four arguments.)